### PR TITLE
fix: filter empty/undefined values  in Multipart

### LINKED ISSUE #10877: [Feature]: Do not send undefined values in multipart
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

When making a JSON request, it would be common to send an `undefined` value for a key. Servers would know to ignore it. For example in a PATCH operation, `undefined` would mean do not make a change, while `null` would indicate to set the value as `null`.

When doing multipart there is no such distinction, as values are either text, either binary. Therefore we should prevent from sending any data for a given key if the value is `undefined`, as sending an empty value for the key (or worse the string `undefined`) does not look like the proper behavior here.

Also currently if the field type is set as `file`, and the value `undefined`. The internal-server would log the following line.
```
The connection observed an error
com.appsmith.external.exceptions.pluginExceptions.AppsmithPluginException: Unable to parse content. Expected to receive an array or object of multipart data
```

### Why should this be worked on?

This would increase the ease of use of the application with multipart APIs

### PR DESCRIPTION
**Pull request by danieldare**
## Description

FIlters out empty/undefined data from the bodyFormData object.

Fixes #10877

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Manually 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feat/undefined-values-in-multipart 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.57 **(0)** | 36.82 **(-0.01)** | 34.86 **(0)** | 55.87 **(0)**
 :green_circle: | app/client/src/sagas/ActionSagas.ts | 12.73 **(0.4)** | 1.49 **(-0.01)** | 8 **(0)** | 15.69 **(0.51)**
 :red_circle: | app/client/src/transformers/RestActionTransformer.ts | 96.77 **(0)** | 85.29 **(-0.82)** | 100 **(0)** | 100 **(0)**</details>

### CODE REVIEW

File: app/client/src/sagas/ActionSagas.ts [hunk 1]
```diff
@@ -308,7 +310,9 @@ export function* updateActionSaga(
     if (!action) throw new Error("Could not find action to update");
 
     if (isAPIAction(action)) {
-      action = transformRestAction(action);
+      // Redux form returns the form values unsantinized, transformRestAction then transforms the action
+      const apiFormAction = yield select(getFormValues(API_EDITOR_FORM_NAME));
```
Comment (AmanAgarwal041) [inline app/client/src/sagas/ActionSagas.ts:314]: Adding a specific form type `API_EDITOR_FOR_NAME` for the form values will make it difficult to track conditions for newer forms. Any reason this was added ?

Comment (danieldare) [reply app/client/src/sagas/ActionSagas.ts:314]: For newer actions we can add for example . `isGRAPHQLAction` to check for specific its use case. 

The new line was added to get the latest form values from the redux form.

**Review by Irongade — DISMISSED** [review]

**Pull request closed without merging**